WireAuthentication/Sources/WireAuthenticationUI/Views/Root/RootViewModel.swift

```diff
@@ -34,7 +34,19 @@ public final class RootViewModel: ObservableObject, Router {
     @Published var modalDestination: RootView.ModalDestination? = .authFlow
     @Published var alert: Alert?
 
-    public init() {}
+    private var cancellable: AnyCancellable?
+    private var lastModalDestination: RootView.ModalDestination?
+
+    public init(bridge: WireAuthenticationBridge) {
+        self.cancellable = bridge.inboundEvents.sink { [weak self] event in
+            switch event {
+            case .didRewindToThisView:
+                self?.restoreSheet()
+            default:
+                break
+            }
```

Comment:
From the registration flow we can go back, and since we dismissed the sheet before starting the registration flow, we need to present it again in case the user goes backwards in the flow. When we add the registration flow to WireAuthentication we can delete these workarounds.

Reply:
question: how long is this RootViewModel kept in memory to keep lastModalDestination? Isn't it the whole WireAuthentication module thrown away once you move to account registration?